serialize/encoder.go

```diff
@@ -52,12 +52,17 @@ import (
 var (
 	byteOrder        binary.ByteOrder = binary.LittleEndian
 	headerMagicBytes                  = encodeUInt16(headerMagicNumber)
+
+	// FOLLOWUP(prateek): this terrible hack is necessary due to a bug in the indexing
+	// sub-system. Our FST library (Vellum) doesn't handle '' values correctly in an edge
+	// case. I'll remove this once I have some cycles to come back to it.
+	mapEmptyTagValueToSpaceHack = ident.BytesID([]byte(" "))
```

Comment:
Can you wrap this in an invalid params error (to return Bad Request)?  `xerrors.NewInvalidParamsError(..)`?